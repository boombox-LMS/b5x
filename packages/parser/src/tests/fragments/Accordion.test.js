const {
  exampleAccordionMarkup,
} = require("../../../dist/componentBuilders/fragments/Accordion");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Accordion fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleAccordionMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed accordion fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed accordion fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
