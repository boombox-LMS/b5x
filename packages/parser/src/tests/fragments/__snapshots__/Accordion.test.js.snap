// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Accordion fragment builds as expected The parsed accordion fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
    "test-fragment-uri",
    "test-fragment-uri",
  ],
  "contentType": "Accordion",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "4d8ba2b09a0224d8f7c939b743d18c47",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
